DSCResources/MSFT_xHostsFile/MSFT_xHostsFile.psm1

```diff
@@ -0,0 +1,314 @@
+$Script:hostsFilePath = "${env:windir}\system32\drivers\etc\hosts"
+
+Import-LocalizedData -BindingVariable LocalizedData -FileName MSFT_xHostsFile.psd1 -BaseDirectory $PSScriptRoot -Verbose
+
+function Get-TargetResource
+{
+    [OutputType([Hashtable])]
+    param (
+        [parameter(Mandatory = $true)]
+        [string]
+        $HostName,
+        [parameter(Mandatory = $true)]
+        [string]
+        $IPAddress
+    )
+
+    $configuration =
+    @{
+        HostName = $HostName
+        IPAddress = $IPAddress
+    }
+
+    Write-Verbose -Message $localizedData.checkingHostsFileEntry
+    try
+    {
+        if (Test-HostEntry -IPAddress $IPAddress -HostName $HostName) {
+            Write-Verbose -Message ($localizedData.hostsFileEntryFound -f $HostName, $IPAddress)
+            $configuration.Add('Ensure','Present')
+        } else {
+            Write-Verbose -Message ($localizedData.hostsFileEntryNotFound -f $HostName, $IPAddress)
+            $configuration.Add('Ensure','Absent')
```

Comment:
Technically it can be present with a different IP address.  In getting, we should return based on the key.